avogadro/core/residue.h

```diff
@@ -0,0 +1,76 @@
+/******************************************************************************
+
+  This source file is part of the Avogadro project.
+
+  Copyright 2018 Kitware, Inc.
+
+  This source code is released under the New BSD License, (the "License").
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+******************************************************************************/
+
+#ifndef AVOGADRO_CORE_RESIDUE_H
+#define AVOGADRO_CORE_RESIDUE_H
+
+#include "avogadrocore.h"
+
+#include <map>
+#include <string>
+
+#include "array.h"
+#include "bond.h"
+#include "vector.h"
+
+namespace Avogadro {
+namespace Core {
+
+/**
+ * @class Residue residue.h <avogadro/core/residue.h>
+ * @brief The Residue class represents a chemical residue.
+ */
+class AVOGADROCORE_EXPORT Residue
+{
+public:
+  /** Type for atom name map. */
+  typedef std::map<std::string, size_t> AtomNameMap;
+
+  /** Creates a new, empty residue. */
+  Residue();
+  Residue(std::string& name);
+  Residue(std::string& name, size_t& number);
+
+  /** Copy constructor  */
+  Residue(const Residue& other);
+
+  /** Assignment operator */
+  Residue& operator=(Residue other);
+
+  /** Destroys the residue object. */
+  virtual ~Residue();
+
+  std::string residueName();
+  void setResidueName(std::string& name);
+
+  size_t residueId();
+  void setResidueId(size_t& number);
+
+  void setResidueAtomName(std::string& name, size_t id);
+  std::vector<std::string> residueAtomNames();
```

Comment:
It would be useful to have a method that returns a vector of the atoms in the residue. For example, this would make it easy to select all atoms in a given residue.